src/services/walletService.ts

```diff
@@ -277,6 +277,10 @@ export default class WalletService {
       return Promise.resolve({ token, allowance: formatUnits(MaxUint256, token.decimals) });
     }
 
+    if (addressToCheck === NULL_ADDRESS) {
```

Comment:
What's the reason behind this check?

Reply:
For wrapping/unwrapping ETH/WETH 0x and other dexes will return the null address as the allowanceTarget since for wrapping/unwrapping you need to call the WETH contract directly and does not need any approval